gui/elements/_viewer.py

```diff
@@ -349,57 +228,61 @@ def _update_statusBar(self):
                     msg = msg + '%d' % value
                 else:
                     msg = msg + '%.3f' % value
+        self._status = msg
+        self.emitStatus()
 
-        self._window._qt_window.statusBar().showMessage(msg)
+    def emitStatus(self):
+            self.statusChanged.emit(self._status)
 
     def on_mouse_move(self, event):
         """Called whenever mouse moves over canvas.
         """
-        if event.pos is None:
-            return
+        if self.layers:
```

Comment:
I'm confused. Are we still in the `Viewer` class? Does the viewer know about annotation layers vs image layers? I thought we wanted it to be blind to the kinds of layers that come in? (This may be impossible with respect to the annotation mode, but I just want to make sure I understand the current architecture.)

Reply:
Yes - this is a problem - I want to pull more of this code out of the viewer and back onto the layers. I plan to do this in a future PR

Reply:
The viewer should pass events to the layers, and it is the business of the layers to know what to do with these events. Better: layers register which canvas events they want to be notified about...

Reply:
Yup that is something we can move towards